### PR TITLE
S6 add sidebar

### PR DESCRIPTION
**Pull request by APaltrow**
Task S6 - Products page - sidebar

- Added UI and adaptivity
- Added categories filter
- Added brands filter
- Added rating filter
- Added price filter
- Added filters reset
- Made filters work with search
- Connected with 'categories' in header

### CODE REVIEW

File: src/components/Products/ProductsContainer/Products.tsx [hunk 1]
```diff
@@ -27,11 +61,16 @@ export const Products: FC = () => {
...
         </div>
       </div>
-
-      <ProductsList
-        productsList={products}
-        isLoading={isLoading}
-      />
+      {isTablet ? (
+        <div className={style.sort_container}>{filterButton}</div>
+      ) : null}
```
Comment (Yuliia143) [inline src/components/Products/ProductsContainer/Products.tsx]: You can simplify this condition

File: src/components/Sidebar/Brands/Brands.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  return (
+    <SidebarBlock title="Brands">
+      <ul className={style.list}>
+        {brandsList.map((brand, idx) => (
+          <li key={`brands_${brand}_${idx + 1}`}>
+            <Checkbox
+              id={brand}
+              isChecked={activeBrands.includes(brand)}
```
Comment (Yuliia143) [inline src/components/Sidebar/Brands/Brands.tsx]: Logic in JSX

File: src/components/Sidebar/SidebarRating/SidebarRating.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    <SidebarBlock title="Rating">
+      <ul className={style.list}>
+        {ratingsList.map((ratingitem, idx) => {
+          const ratingValue = ratingsList.length - idx;
+
+          return (
+            <li key={`${ratingitem}_${ratingValue}`}>
+              <Checkbox
+                id={`checkbox_${ratingValue}`}
+                isChecked={activeRatings.includes(ratingValue)}
```
Comment (Yuliia143) [inline src/components/Sidebar/SidebarRating/SidebarRating.tsx:37]: Logic in JSX

File: src/components/ui/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        className={style.input}
+      />
+      <label
+        htmlFor={id}
+        className={style.label}
+      >
+        <span className={style.content}>{children}</span>
+      </label>
+
+      {isChecked ? (
```
Comment (Yuliia143) [inline src/components/ui/Checkbox/Checkbox.tsx]: Can simplify condition

File: src/components/ui/Slider/Slider.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+}) => {
+  const positionLeft =
+    sliderMin - MIN_GAP > ZERO_POSITION && !isDisabled
+      ? sliderMin - MIN_GAP
+      : ZERO_POSITION;
+
+  const positionRight =
+    sliderMax - MIN_GAP > ZERO_POSITION && !isDisabled
+      ? sliderMax - MIN_GAP
+      : ZERO_POSITION;
```
Comment (Yuliia143) [inline src/components/ui/Slider/Slider.tsx]: You could create a function and pass sliderMax or sliderMin to calculate values

File: src/helpers/filterProducts.ts [hunk 6]
```diff
@@ -11,6 +11,23 @@ export const filterProducts = {
...
   },
+  byRatings: (products: IProduct[], activeRatings: number[]) => {
+    if (!activeRatings.length) return products;
+
+    return products.filter(({ rating }) => activeRatings.includes(rating));
+  },
+  byPriceRange: (products: IProduct[], priceRange: IPriceRange) => {
+    const { min, max } = priceRange;
+
+    if (min === 0 && max === 0) return products;
```
Comment (Yuliia143) [inline src/helpers/filterProducts.ts]: Simplify ?

File: src/components/Products/ProductsContainer/Products.tsx (hunk 1 above)
Comment (APaltrow) [reply src/components/Products/ProductsContainer/Products.tsx]: Removed ternary condition

File: src/components/Sidebar/Brands/Brands.tsx (hunk 2 above)
Comment (APaltrow) [reply src/components/Sidebar/Brands/Brands.tsx]: Set logic to a separate function

File: src/components/ui/Slider/Slider.tsx (hunk 5 above)
Comment (APaltrow) [reply src/components/ui/Slider/Slider.tsx]: Set logic to a separate function

File: src/helpers/filterProducts.ts (hunk 6 above)
Comment (APaltrow) [reply src/helpers/filterProducts.ts]: Removed comparison with 0

**Review by Yuliia143 — APPROVED** [review]

File: src/components/ui/NoResults/NoResults.tsx [hunk 7]
```diff
@@ -10,8 +10,10 @@ export const NoResults: FC = () => {
     <article className={style.container}>
       <h3>No Results Found</h3>
       <p>
-        Could not find a match for
-        <span className={style.search_text}>{` '${searchValue}'`}</span>
+        Could not find a match
+        {searchValue ? (
+          <span className={style.search_text}>{` for '${searchValue}'`}</span>
+        ) : null}
```
Comment (vynnykpff) [inline src/components/ui/NoResults/NoResults.tsx]: Try to simplify this condition, using **&&**

**Review by vynnykpff — APPROVED** [review]

File: src/components/ui/NoResults/NoResults.tsx (hunk 7 above)
Comment (APaltrow) [reply src/components/ui/NoResults/NoResults.tsx]: Substituted ternary with &&

**Pull request merged**